### DISCUSSION TITLE
TestFlight for Buzz Captions

### DISCUSSION THREAD

**Discussion by chidiwilliams** [chidiwilliams/buzz#374] [Show and tell]
I've been working on an improved version of Buzz on macOS. Here's a test flight link: https://testflight.apple.com/join/0T5byiDM

![image](https://user-images.githubusercontent.com/17879672/224502213-a0ddf266-31ae-4bd2-94ca-6648bd2b7861.png)

**Comment by chidiwilliams** [comment]
@Sircam19, want to help test?

**Reply by Sircam19** [reply]
So I tested a few things really sorry I couldn't send screen shots right now. 

1. I didn't download a model prior to importing a transcription so this may be something to make crystal clear for other users.

2. After loading the medium model AND importing a file for transcription there was "activity" but no transcription even after 10 minutes. Perhaps a progress monitor would be useful for down the road.
3. After restarting the app (cmd Q)and trying another transcription (albeit with same mp3 file) the previous task still seemed to be running.
4. I'll try another time once kids are napping and I have a stable connection at the resort👍🏿

**Reply by Sircam19** [reply]
5. Even after doing two force quits and then reopening the app it's still stuck on that previous transcription.

**Reply by Sircam19** [reply]
6. Language options in pickable list does not include language recognition option even when downloading the medium non-en version.

**Reply by chidiwilliams** [reply]
- Yes, I'm already working on a progress monitor, but could you try with a smaller model and a smaller audio file?
- Will add the language recognition option in a bit.

Thanks for the feedback. I'll have an update out soon.

**Reply by Sircam19** [reply]
7. Kind Sir. So remotes back into my machine and unfortunately it's still trying to transcribe. It's been from almost yesterday.
![032F0F05-8AC2-4B00-948A-F9306B4F80AE](https://user-images.githubusercontent.com/4651001/224562909-84682ff2-7cb8-4346-93eb-f1cdb34f29db.jpeg)

**Reply by Sircam19** [reply]
> * Yes, I'm already working on a progress monitor, but could you try with a smaller model and a smaller audio file?
> * Will add the language recognition option in a bit.
> 
> Thanks for the feedback. I'll have an update out soon.

Thanks Chidi will do. Am using my phone to reply and missed your responses. Yes will attempt again using tiny model...apologies didn't see your response prior to sending image. Thank you.

**Reply by Sircam19** [reply]
Hello good morning. Unfortunately same issue as described before using Tiny model. Regards.

**Reply by chidiwilliams** [reply]
Hey, thanks. Could you try the new update? You should have it on your TestFlight app.

**Reply by Sircam19** [reply]
I tried the latest and still getting a few crashes even under the tiny model. After a crash I have the option to restart which restarts the app fine but after I add a new transcription it still takes for ever but without an output.

**Reply by chidiwilliams** [reply]
Thanks for the feedback. That's pretty surprising. Could you try this:

- Stop and delete all the currently running transcriptions. (You can right-click on a transcription / click delete to delete it)

  ![image](https://user-images.githubusercontent.com/17879672/224834272-a9e3bab6-3fc9-443f-abe0-230ea0eaebfc.png)

- Restart the app and check that it starts out empty
- Then, try to transcribe this file (https://github.com/chidiwilliams/buzz/blob/main/testdata/whisper-french.mp3) with the tiny model.

Does that work for you?

**Reply by chidiwilliams** [reply]
Think I may have found the issue actually, but let me know if that file works for you.

**Reply by Sircam19** [reply]
I did try the update… It wants to transcribe but no output as of yet :-(

**Reply by Sircam19** [reply]
So I tried again following exactly your steps especially noting the test file you provided but no luck. It was really hard having it start empty and when I did have it start empty I got the following. 
![Uploading 6F20A5B2-5E2C-4686-913E-F608C8EC7D29.png…]()

**Reply by chidiwilliams** [reply]
Looks like your image didn't send correctly. Could you try it again?

**Reply by Sircam19** [reply]
> Looks like your image didn't send correctly. Could you try it again?

![Uploading 506885B3-1E8C-4745-9952-23C7E11B2761.png…]()

**Reply by Sircam19** [reply]
![Screenshot 2023-03-14 at 6 35 10 AM](https://user-images.githubusercontent.com/4651001/224989334-63e608d1-a55f-400b-9369-55a728dc4183.png)

**Reply by Sircam19** [reply]
Here it is sorry about that 👍🏿 I installed the swift version on my older intel Mac as lost ability to access my M1 remotely but that shouldn’t be part of the issue or could it be?

**Reply by chidiwilliams** [reply]
Got a new update out. If you have some time, could you try again. Really appreciate the help. The app should work on both Intel and M1.

**Reply by Sircam19** [reply]
> Got a new update out. If you have some time, could you try again. Really appreciate the help. The app should work on both Intel and M1.

![Screenshot 2023-03-14 at 6 39 41 PM png](https://user-images.githubusercontent.com/4651001/225166390-3ed47ce9-a8a3-476d-8414-54e24e7ea704.png)
![Screenshot 2023-03-14 at 6 42 26 PM png](https://user-images.githubusercontent.com/4651001/225166393-c0a63f4c-a819-48c0-a401-bc4a48c03ecc.png)

**Reply by Sircam19** [reply]
Tried the provided audio file and one of my own. Provided audio sample on Tiny model and the second audio file under medium.

**Reply by Sircam19** [reply]
> Tried the provided audio file and one of my own. Provided audio sample on Tiny model and the second audio file under medium.

Both were mp3 files so perhaps the models are expecting a WAV file but not sure.

**Reply by chidiwilliams** [reply]
Okay, perfect. This is good progress. I get where the issue was from now, but I'm still not sure why it's failing on your end. I'll get back to you shortly.

**Reply by Sircam19** [reply]
> Okay, perfect. This is good progress. I get where the issue was from now, but I'm still not sure why it's failing on your end. I'll get back to you shortly.

All good I’ll have access to my M1 machine from Friday and whenever I have a spare moment I’d be honoured to help out 👍🏿

**Reply by chidiwilliams** [reply]
Great, thanks. There's a new version out now.

**Reply by Sircam19** [reply]
I feel like it’s getting closer but didn’t recognize the mp3 file
![Uploading Screenshot 2023-03-15 at 10.03.20 PM.png.png…]()

**Reply by Sircam19** [reply]
![Screenshot 2023-03-15 at 10 05 31 PM png](https://user-images.githubusercontent.com/4651001/225501895-6a67ebf9-8dae-40a1-ac44-d94a85e20b58.png)

**Reply by chidiwilliams** [reply]
There's a new build 🤞🏾

**Reply by Sircam19** [reply]
Okay success it got much much further than before although the French mp3 file yielded 2 lines of text under the medium model but this is a great milestone… woohooo. I’ll try some other audio files between flights while I can before heading home 👍🏿 Great news.
![Screenshot 2023-03-16 at 3 02 13 PM png](https://user-images.githubusercontent.com/4651001/225740040-3a9aa818-96f7-45ea-84ab-7cf9f3e67016.png)

**Reply by Sircam19** [reply]
Loving the time remaining progress indicator. It seems to be working well. 👍🏿
![Screenshot 2023-03-16 at 3 10 58 PM png](https://user-images.githubusercontent.com/4651001/225742052-d640c14b-dc24-4212-bade-f684784007ae.png)

**Reply by Sircam19** [reply]
Success outputted perfectly… Well done Chidi …well done👍🏿🎉🎉👍🏿

**Reply by chidiwilliams** [reply]
yes the test mp3 file should only have two lines of audio.

**Reply by chidiwilliams** [reply]
Great, that's amazing to hear 🎉. Btw, pro tip: you can double click on the title and the transcriptions to edit them :)

**Comment by Huansock** [comment]
Can i take a Test too? i am korean and m1 mac air user

**Reply by chidiwilliams** [reply]
Sure, just join with the link

**Comment by lantc77** [comment]
<img width="889" alt="截屏2023-03-15 12 28 17" src="https://user-images.githubusercontent.com/27905819/225206626-b3a7f122-81b9-4cad-828b-42e2ccc41da4.png">

For me, the test buzz is always "processing" the 1-minute mp3 file even after I have waited for minutes. This didn't happen occasionally, I have tried other mp3 files but they all failed. Thanks.

**Reply by lantc77** [reply]
By the way, the device is Macbook pro 2021 (M1 pro chip)

**Reply by lantc77** [reply]
And I find that I can't delete the files even after I force to quit it or reboot Mac, every time I open the app the same files occur.

**Reply by chidiwilliams** [reply]
Could you try updating to the latest version?

**Reply by lantc77** [reply]
Thank you, after installing the latest version, it worked in processing data. But the problem of deleting files still exists, every time I open the app the deleted files occur again.

**Reply by Sircam19** [reply]
> Thank you, after installing the latest version, it worked in processing data. But the problem of deleting files still exists, every time I open the app the deleted files occur again.

Excellent so by processing the file you got a transcript? Awesomeness, hasn’t worked for me as of yet

**Reply by chidiwilliams** [reply]
The transcripts should be stored in `/Users/username/Library/Containers/com.chidiwilliams.Buzz/Data/Library/Application Support/Buzz/transcriptions.data`. Could you try deleting the file and restarting the app?
